commons-app/src/main/java/cz/cuni/mff/xrg/odcs/commons/app/auth/DefaultPermissionEvaluator.java

```diff
@@ -60,48 +93,88 @@ public boolean hasPermission(Authentication auth, Object target, Object perm) {
         // check if our entity is shared
         if (target instanceof SharedEntity) {
             SharedEntity sTarget = (SharedEntity) target;
-            switch (perm.toString()) {
-                case "view":
-                case "use":
-                case "copy":
-                case "export":
-                    if (ShareType.PUBLIC.contains(sTarget.getShareType())) {
-                        return true;
-                    }
-                    break;
-                case "save":
-                    if (ShareType.PUBLIC_RW.equals(sTarget.getShareType())) {
+            if (foundPermission != null) {
+                if (foundPermission.isRwOnly() && !ShareType.PUBLIC_RW.equals(sTarget.getShareType())) {
```

Comment:
there are permissions which can be evaluated to true ONLY if object is in
the RW visibility state, in other words user which has permission xxx.edit
can edit his own objects and objects which are visible to others in rw
mode. that is the meaning of extra column in the permission table

On Sun, Mar 22, 2015 at 11:32 AM, Tomas Knap <notifications@github.com>
wrote:

> In
> commons-app/src/main/java/cz/cuni/mff/xrg/odcs/commons/app/auth/DefaultPermissionEvaluator.java
> <https://github.com/UnifiedViews/Core/pull/282#discussion_r26901549>:
>
> > @@ -60,48 +93,88 @@ public boolean hasPermission(Authentication auth, Object target, Object perm) {
> >          // check if our entity is shared
> >          if (target instanceof SharedEntity) {
> >              SharedEntity sTarget = (SharedEntity) target;
> > -            switch (perm.toString()) {
> > -                case "view":
> > -                case "use":
> > -                case "copy":
> > -                case "export":
> > -                    if (ShareType.PUBLIC.contains(sTarget.getShareType())) {
> > -                        return true;
> > -                    }
> > -                    break;
> > -                case "save":
> > -                    if (ShareType.PUBLIC_RW.equals(sTarget.getShareType())) {
> > +            if (foundPermission != null) {
> > +                if (foundPermission.isRwOnly() && !ShareType.PUBLIC_RW.equals(sTarget.getShareType())) {
>
> What is this magical row doing - can you please explain idea behind? This
> relates to rather not wanted extra column in permissions table.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/UnifiedViews/Core/pull/282/files#r26901549>.
>